src/main/java/org/isf/menu/service/MenuIoOperations.java

```diff
@@ -313,30 +305,32 @@ private GroupMenu insertGroupMenu(UserGroup aGroup, UserMenuItem item) throws OH
 
 	/**
 	 * Deletes a {@link UserGroup}
-	 * 
 	 * @param aGroup - the {@link UserGroup} to delete
 	 * @throws OHServiceException When failed to delete group
 	 */
 	public void deleteGroup(UserGroup aGroup) throws OHServiceException {
-		groupMenuRepository.deleteWhereUserGroup(aGroup.getCode());
-		groupPermissionIoOperationRepository.deleteAllByUserGroup_Code(aGroup.getCode());
-		groupRepository.delete(aGroup);
+		ensureUserGroupNotDeleted(aGroup.getCode());
+		if (getUser(aGroup.getCode()).isEmpty()) {
+			groupRepository.delete(aGroup);
+		} else {
+			aGroup.setDeleted(true);
+			groupRepository.save(aGroup);
+		}
```

Comment:
In this way a group will be set to deleted only when there are no users in it, but in this case why to keep it in the DB?

Reply:
sorry, misunderstood because I'm having an error when deleting not empty groups, so maybe is for another reason

Reply:
the check about existing users are already done in the manager... maybe we have not to change the logic of the `deleteGroup` (or `deleteUser`) methods in IoOperations as you did, but just use the update methods to set the `deleted=true`

Reply:
in other words, hard-deletion should use "delete" methods (and return void) soft-deletion is done with a "update" methods